File: poc-kronos-frontend/src/model/curso/curso-funcionario/CursoFuncionario.js
class CursoFuncionario {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataInicio = dados ? dados.dataInicio : null;
    this.dataFinal = dados ? dados.dataFinal : null;
    this.idFuncionario = dados ? dados.idFuncionario : null;
    this.nomeFuncionario = dados ? dados.nomeFuncionario : null;
    this.idCurso = dados ? dados.idCurso : null;
    this.idTipoFuncao = dados ? dados.idTipoFuncao : null;
    this.nomeTipoFuncao = dados ? dados.nomeTipoFuncao : null;
  }

  static validations(required, dataFimMenor) {
    return {
      dataInicio: { required },
      dataFinal: { dataFimMenor: dataFimMenor("dataInicio") },
      idFuncionario: { required },
      idCurso: { required },
      idTipoFuncao: { required }
    };
  }

  static validationMessages() {
    return {
      dataInicio: {
        required: "Data Início é obrigatório"
      },
      dataFinal: {
        dataFimMenor: "Data Final não pode ser menor que a Data Início"
      },
      idFuncionario: {
        required: "Funcionário é obrigatório"
      },
      idCurso: {
        required: "Curso é obrigatório"
      },
      idTipoFuncao: {
        required: "Tipo de Função é obrigatória"
      }
    };
  }
}
export default CursoFuncionario;

File: src/main/java/br/com/kronos/backend/aplicacao/curso/CursoFuncionario.java
package br.com.kronos.backend.aplicacao.curso;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class CursoFuncionario {
    
    private long id;
    private LocalDate dataInicio;
    private LocalDate dataFinal;
    private Long idFuncionario;
    private Long idCurso;
    private Integer idTipoFuncao;

	@Builder
	public CursoFuncionario(long id, LocalDate dataInicio, LocalDate dataFinal, Long idFuncionario, Long idCurso, Integer idTipoFuncao) {

		super();
        this.id = id;
        this.dataInicio = exigirNaoNulo(dataInicio, "Data de início");
        this.dataFinal = dataFinal;
        this.idFuncionario = exigirNaoNulo(idFuncionario, "Funcionário");
        this.idCurso = exigirNaoNulo(idCurso, "Curso");  
        this.idTipoFuncao = exigirNaoNulo(idTipoFuncao, "Tipo de função");       
	}
}
